### ISSUE TITLE
Python SDK: `API.get_archived_experiments` returns non-archived experiments + `RestApiClient.get_project_experiments` payload does not include metadata from archived experiments

### ISSUE THREAD

**Issue by nathanpainchaud** [comet-ml/issue-tracking#508]
### `API.get_archived_experiments` returns non-archived experiments
In the Python SDK, methods for accessing archived experiments (e.g. `API.get_archived_experiments`) provide wrong lists of experiments in at least some cases.  
  
When providing a project name but no `pattern` argument to `API.get_archived_experiments`, the method returns the list of non-archived experiments in the project, i.e. the exact same list as `API.get_experiments`.
  
This happens because the call to `API._gen_experiments` falls in the `elif pattern is None` case, which calls `API._get_project_experiments` without passing the `archived` parameter, which therefore defaults to `False` and returns non-archived experiments. `archived=archived` should thus be added to the call to `API._get_project_experiments` in that case as well.

Below the fix that should be done:
```diff
        elif pattern is None:
-           experiments = self._get_project_experiments(workspace, project_name)
+           experiments = self._get_project_experiments(workspace, project_name, archived=archived)
            for metadatum in experiments.values():
                yield APIExperiment(api=self, metadata=metadatum)
            return
```

### `RestApiClient.get_project_experiments` payload does not include metadata from archived experiments
After manually fixing the bug described above, I tried to test `API.get_archived_experiments` again, which this time returns an empty list, even though I have archived experiments in my project. I noticed that it was because, in the call to `API._get_project_experiments` detailed above, the `project_json` dictionary returned by the call to `API._client.get_project_experiments` (i.e. the wrapper around the call to the REST API) only contains the metadata of non-archived experiments. Therefore, once the filter for archived experiments is applied (i.e. the dictionary comprehensions with the `metadata["archived"] == archived` condition), then we're left with empyt dicts/lists.  
  
At this point, I gave up trying to understand _why_ `API._client.get_project_experiments` doesn't include the metadata of archived experiments, since the problem seems to no longer be with the Python SDK but rather the REST API (which I didn't want to begin digging into).

The call to the `RestApiClient` that lacks metadata from archived experiments:
```python
# Get the project details:
project_json = self._client.get_project_experiments(workspace, project_name)
if project_json is None:
    return
```
and the dict comprehension that causes the experiments dict to become empty (because no experiments included in the `project_json` payload are archived):
```python
experiments = {
    metadata["experimentKey"]: metadata
    for metadata in project_json["experiments"]
    if metadata["archived"] == archived
}
```

## Expected behavior
Calling `API.get_archived_experiments` (for a specific workspace and project) should return the list of archived experiments.

## Where is the issue?
- [x] Comet Python SDK
- [X] Comet REST API
- [ ] Comet UI
- [ ] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lighting etc)

## To Reproduce
Read the detailed description of the bug.

**Comment by dsblank** [comment]
I believe that this has been fixed. If you have any issues, please re-open. Thank for reporting!

**Comment by nathanpainchaud** [comment]
@dsblank Thanks for responding to this issue.

> I believe that this has been fixed. 

I did a quick test on a project I have which contains archived experiments, and I found that:

- The issues described originally are indeed fixed, as in some archived experiments do get picked up now;
- However, the number of experiments I got back when querying for all experiments (normal + archived) experiments is bigger than the total number of experiments (normal + archived) in my project...

Therefore, the API to query archived experiments still does not behave as expected. I don't have the time to properly investigate this right now to give more info about what experiments get picked up when they shouldn't (or get picked up twice?).

> If you have any issues, please re-open. Thank for reporting!

I figure that since this would be a new bug in itself, you'd want to move this discussion to a new issue to track it more easily?
